app/src/main/java/com/owncloud/android/utils/FileExportUtils.kt

```diff
@@ -0,0 +1,156 @@
+                } else if (file != null) {
+                    FileInputStream(file)
+                } else {
+                    throw RuntimeException("ocFile and file both may not be null")
+                }
+
+                val pfd = contentResolver.openFileDescriptor(uri, "w")
+                val outputStream = FileOutputStream(pfd!!.fileDescriptor)
+                IOUtils.copy(inputStream, outputStream)
+            } catch (e: IOException) {
+                e.printStackTrace()
+            }
+        } else {
+            // < Android 10
+            var outputStream: FileOutputStream? = null
+            var inputStream: InputStream? = null
+            try {
+                var target = File(
+                    Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS),
+                    fileName
+                )
+
+                if (target.exists()) {
+                    var count = INITIAL_RENAME_COUNT
+                    do {
+                        val name = generateNewName(fileName, count)
+                        target = File(
+                            Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_DOWNLOADS),
+                            name
+                        )
+
+                        count++
+                    } while (target.exists())
+                }
+
+                inputStream = if (ocFile != null) {
+                    contentResolver.openInputStream(ocFile.storageUri)
+                } else if (file != null) {
+                    FileInputStream(file)
+                } else {
+                    throw RuntimeException("ocFile and file both may not be null")
+                }
+
+                outputStream = FileOutputStream(target)
+
+                inputStream.use { fis ->
+                    outputStream.use { os ->
+                        val buffer = ByteArray(1024)
+                        var len: Int
+                        while (fis!!.read(buffer).also { len = it } != -1) {
+                            os.write(buffer, 0, len)
+                        }
+                    }
+                }
+            } catch (e: FileNotFoundException) {
+                e.printStackTrace()
+            } catch (e: IOException) {
+                e.printStackTrace()
+            } finally {
+                outputStream?.close()
+                inputStream?.close()
+            }
+        }
+    }
+
+    private fun generateNewName(name: String, count: Int): String {
+        val extPos = name.lastIndexOf('.')
+        val suffix = " ($count)"
+        var extension = ""
+        var nameWithoutExtension = ""
+
+        if (extPos >= 0) {
+            extension = name.substring(extPos + 1)
+            nameWithoutExtension = name.substring(0, extPos)
+        }
+
+        return if (extPos >= 0) {
+            "$nameWithoutExtension$suffix.$extension"
+        } else {
+            name + suffix
+        }
```

Comment:
```suggestion
        val extPos = name.lastIndexOf('.')
        val suffix = " ($count)"

        return if (extPos >= 0) {
            val extension = name.substring(extPos + 1)
            val nameWithoutExtension = name.substring(0, extPos)
            "$nameWithoutExtension$suffix.$extension"
        } else {
            name + suffix
        }
```
Can be simplified and made immutable by merging the `if`s